### ISSUE TITLE
nothing showing on window

### ISSUE THREAD

**Issue by 3ux1n3** [TomSchimansky/CustomTkinter#1690]
system : macos ventura

code : 

```
import customtkinter

def button_callback():
    print("button pressed")

app = customtkinter.CTk()
app.title("my app")
app.geometry("400x150")

button = customtkinter.CTkButton(app, text="my button", command=button_callback)
button.grid(row=0, column=0, padx=20, pady=20)

app.mainloop()
```

when i run this i can't see anything in window

screenshot : [https://imgur.com/a/01xLBKg](https://imgur.com/a/01xLBKg)

**Comment by avalon60** [comment]
Seems ok to me:

![image](https://github.com/TomSchimansky/CustomTkinter/assets/89534395/2e8ce3b5-8dfa-4355-9479-d0433ead542a)

**Comment by iiasceri** [comment]
@3ux1n3 arent they retarded xD ? same thing my dude same thing ventura

**Comment by iiasceri** [comment]
When I see issues like these I can't be serious about using this crap library/framework, the only reason I am not using kmm and want python is cuz the other crap doesnt work in kmm like I wanted to do an ocr which python did just fine and in kmm ml kit is android/ios only I did desktop it works now gui missing. Now u get why the folks made apis etc? in my case I can just expose that as a server and make gui in something decent either kmm or native macos (xcode ide for idiots), native windows with their idiotic ide as well studio bs. Jetbrains are amazing they used their talent to make at least kmm its just when it comes to truly cross platform it is always cancer at somepoint like u experienced with nothing displaying.
The fact that u even bothered opening this issue is a good thing from ur side but stop it these devs are just ... ugh... Just drop programming xD do some gardening, enjoy beauty of a sunset a company of a friend if u have. I am done with tech :)

**Comment by 3ux1n3** [comment]
> You aren't expecting the text "button pressed" to appear inside the window are you? If you are then that is a misconception.

no i am expecting the button to be visible on the window

**Comment by duruburak** [comment]
Try updating `customtkinter`. Works fine for me.
`pip install --upgrade customtkinter`

**Comment by Akascape** [comment]
@3ux1n3 Very rare case. Please try this basic example code: https://github.com/TomSchimansky/CustomTkinter/blob/master/examples/simple_example.py

If it works then it is a problem in your previous code (or maybe the file). 

Otherwise it can be a problem with your mac os tkinter. Tkinter is little buggy on mac systems.

**In that case try updating python.**

**Comment by jocelynmass** [comment]
I had the same issue with Ventura and Python3.9.6. I fixed the issue by creating a virtual env using python3.11:
here are the commands I ran:

```
$ virtualenv --python="/opt/homebrew/bin/python3.11" venv 
$ source venv/bin/activate 
$ pip install --upgrade pip 
$ pip install customtkinter
$ brew install python-tk
```

**Comment by itsPreto** [comment]
thanks @jocelynmass worked like a charm! Though I think there should be a note in the README to warn users of this-- I spent a whole day on this thinking i was doing something wrong..

**Comment by dishb** [comment]
@jocelynmass - I really don't understand why a `venv` is going to fix this issue. Any ideas?

> @3ux1n3 arent they idiots xD ? same thing my dude same thing ventura

@iiasceri - Way to be respectful! (sarcasm). Most Code of Conducts would not allow this kind of behavior. Also doesn't look like you tried to _solve_ the issue. Complaining hardly ever fixes things! :)

> thanks @jocelynmass worked like a charm! Though I think there should be a note in the README to warn users of this-- I spent a whole day on this thinking i was doing something wrong.. @3ux1n3 this is the answer you want

@itsPreto - I don't think we can assume it's a CustomTkinter problem (yet).

@3ux1n3 - I ran the code you provided, the button shows up for me:

![Screen Shot 2023-06-13 at 5 07 45 PM](https://github.com/TomSchimansky/CustomTkinter/assets/130259005/2b292cca-5c6a-4e0f-869d-647e359b0c5a)

Specs:
- macOS 12.6.6
- CustomTkinter 5.1.3
- Python 3.11.3

**Comment by itsPreto** [comment]
@dishb have you tried downgrading your python to see at which point it breaks

**Comment by 3ux1n3** [comment]
i am using pipenv python version annd everything is uptodate , it still not fixed

**Comment by dishb** [comment]
> @dishb have you tried downgrading your python to see at which point it breaks

The [author of this issue said](https://github.com/TomSchimansky/CustomTkinter/issues/1690#issuecomment-1590255375) that, despite updating everything (including Python), the issue is not fixed.

**Comment by dishb** [comment]
> i am using pipenv python version annd everything is uptodate , it still not fixed

Can you try _not_ using a environment?

**Comment by dishb** [comment]
Perhaps this is similar to #1692

**Comment by jocelynmass** [comment]
@dishb I'm not saying `venv` is fixing the problem. I only use it to set Python v3.11 without messing up my general Python environment. The issue seems related to the default Python version used with Ventura. I didn't check what exact version fixes the problem though.

**Comment by 3ux1n3** [comment]
the env i setup is python 3.11 i don’t use the system python binary

**Comment by dishb** [comment]
> @dishb I'm not saying `venv` is fixing the problem. I only use it to set Python v3.11 without messing up my general Python environment. The issue seems related to the default Python version used with Ventura. I didn't check what exact version fixes the problem though.

Yeah I don't think Tkinter supports Python 2... I believe macOS ships with Python 2.9 or 3.9... It's not the latest version and is pretty outdated. Also the reason on Mac's you have to use `python3` or `pip3` commands.

@jocelynmass - Unfortunately, the author (@3ux1n3) [commented](https://github.com/TomSchimansky/CustomTkinter/issues/1690#issuecomment-1590637539) (after you) saying that they are using an environment running Python 3.11.

Seems like the version of Python is _not_ the issue.

**Comment by dishb** [comment]
@3ux1n3 - What version of CustomTkinter are you using? I think version of macOS is irrelevant here.

**Comment by itsPreto** [comment]
@TomSchimansky I just tried this out of curiosity and with the standard Tkinter and python 3.9 I'm still seeing the issue-- but the moment I upgrade python to 3.11 it's able to render the button.

**Comment by jocelynmass** [comment]
Actually, @3ux1n3 sharing with us your python config could be helpful: `pip3 freeze > requirements.txt`

**Comment by 3ux1n3** [comment]
this issue was fixed by making sure pipenv uses python 3.11+ to build the virtual environment it was defaulting to system python on mac

**Comment by dishb** [comment]
@3ux1n3 - Great! So it was the version of Python...
